src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -89,10 +89,19 @@ internal abstract class AbstractPullDiagnosticHandler<TDiagnosticsParams, TRepor
         protected abstract ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(RequestContext context, CancellationToken cancellationToken);
 
         /// <summary>
-        /// Creates the <see cref="VSInternalDiagnosticReport"/> instance we'll report back to clients to let them know our
-        /// progress.  Subclasses can fill in data specific to their needs as appropriate.
+        /// Creates the appropriate LSP type to report a new set of diagnostics and resultId.
         /// </summary>
-        protected abstract TReport CreateReport(TextDocumentIdentifier identifier, LSP.Diagnostic[]? diagnostics, string? resultId);
+        protected abstract TReport CreateReport(TextDocumentIdentifier identifier, LSP.Diagnostic[] diagnostics, string resultId);
```

Comment:
result of bug fixing - for VSCode diagnostics we were previously creating unchanged reports for removed documents.  This splits it out so we know exactly which is being called and can create appropriate types